src/mgmt/srv_query.c

```diff
@@ -41,58 +41,60 @@ bs_state_query(void *arg)
 	bio_get_bs_state(&bs_arg->bs_arg_state, bs_arg->bs_arg_type, bxc);
 }
 
+static inline int
+tgt2xs_type(int tgt_id)
+{
+	return tgt_id == BIO_SYS_TGT_ID ? DSS_XS_SYS : DSS_XS_VOS;
+}
+
 /*
  * CaRT RPC handler for management service "get blobstore state" (C API)
  *
  * Internal blobstore states returned for test validation only.
  */
 int ds_mgmt_get_bs_state(uuid_t bs_uuid, int *bs_state)
 {
-	struct smd_dev_info		*dev_info;
-	ABT_thread			 thread;
-	int				 tgt_id;
-	int				 rc;
-	struct bs_state_query_arg	 bs_arg;
+	ABT_thread			thread;
+	int				tgt_id;
+	int				rc;
+	struct bs_state_query_arg	bs_arg;
+	int				xs_type;
 
-	bs_arg.bs_arg_type = SMD_DEV_TYPE_DATA;
 	/*
 	 * Query per-server metadata (SMD) to get target ID(s) for given device.
 	 */
 	if (!uuid_is_null(bs_uuid)) {
-		rc = smd_dev_get_by_id(bs_uuid, &dev_info);
-		if (rc != 0) {
-			D_ERROR("Blobstore UUID:"DF_UUID" not found\n",
+		tgt_id = bio_dev_owner_xs_id(bs_uuid);
```

Comment:
The device list in nvme_glb.bd_bdevs is managed by init xstream, it's not thread safe, though the race window is small (the list is only changed on xstream start or hotplug), we just can iterate the list from other xstreams.

I know that there might be existing code having similar problems, maybe we'd use this opportunity to cleanup and fix all potential issues.